### PR TITLE
Turns Perf into The Standard Oxychem, Heavily Tweaks Perf To Fit My Freshly Repackaged Fantasy

### PR DESCRIPTION
**Pull request by ExcessiveUseOfCobblestone**
### Salbu is No Longer The Standard
- Salbu has been removed from every roundstart machinery. It was way too good as a roundstart chem. It can still be crafted.
- Salbu has been added to sleepers as a T3 upgrade chemical (hehe)

### Perf's Debut
- Perf has replaced every entry where Salbu was. Perf converts O2 healing into lesser amounts of Toxin damage more evenly than previous (40 : 1 [lol] to 5 : 2)
- Perf now ramps up effects the longer you have it in your system (`current_cycle`)
- You will always get at least 0.5 tox damage a cycle even without O2 healing to avoid doping.
- Perf now has a 35u OD. The OD mechanic is the same as it functions normally but does NOT account for actual O2 healed. The threshold rational was to ensure one needed at least 3 syringes from a syringe gun to OD.

### Tweaks
- Salbu's 50u pills were replaced with Perf 15u syringes. This is because Perf ramps so you won't need as much.
- The O2 Deprivation kit dropped a pen since It's not a "critkit" but an O2 kit and because epi compliments perf. Make your own Epi if you need another pen.
- TB will now use Perf to cure

## Why It's Good For The Game

All in all my goal here is to make oxygen damage something that is a little less forgiving without more preparation. I did this by replacing the standard BTFO2 chem with a lesser chem. However, the "lesser chem" was initially arms racing with the standard (that shouldn't have been the standard) so I just tweaked it while keeping it conceptually similar (high o2 healing with toxin conversion).

#### Salbu Removal
This is because Salbu was both more than capable of healing you out of any situation O2 wise and permitted doping, while also being the most abundant O2 chem.

#### Perf's Rework
This is because Perf was borderline OP pre-PR compared to the MUCH HARDER to craft Salbu. I wanted to rework it to be more of an emergency chem, with emergency meaning it comes at a cost that wasn't laughably negligible and you wanted to be conscious about how much you're putting into someone.

## Changelog
:cl: cO2by
del: Salbu has been replaced in all roundstart items with the PERFect chem
add: Salbu is now a T3 sleeper upgrade
balance: Perf now converts O2 deprivation to lesser amounts of Toxin more evenly than before
add: Perf effects scale based on O2 deprivation and how long perf is in your system
balance: Perf now has a 35u OD which ignores O2dep adjustment (You will take scaled toxin damage regardless of healed oxygen damage)
/:cl:

### CODE REVIEW

**Comment by ExcessiveUseOfCobblestone** [comment]
wrong PR Desc 1 sec

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 1]
```diff
@@ -505,17 +506,27 @@
...
+	overdose_threshold = 35 // at least 2 full syringes +some, this stuff is nasty if left in for long
 
 /datum/reagent/medicine/perfluorodecalin/on_mob_life(mob/living/carbon/human/M)
-	M.adjustOxyLoss(-12*REM, 0)
-	M.adjustToxLoss(0.3*REM, 0)
+	var/oxycalc = 2.5*REM*current_cycle
+	if(!overdosed)
+		oxycalc = min(oxycalc,M.getOxyLoss()+PERF_BASE_DAMAGE) //if NOT overdosing, we lower our toxdamage to only the damage we actually healed with a minimum of 0.5. IE if we only heal 10 oxygen damage but we COULD have healed 20, we will only take toxdamage for the 10. We would take the toxdamage for the extra 10 if we were overdosing.
+	M.adjustOxyLoss(-oxycalc, 0)
+	M.adjustToxLoss(oxycalc/2.5, 0)
```
Comment (Tlaltecuhtli) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: so every tick it increases the tox damage dealt on each tick by 1? doesnt this mean that 5u is lethal?

**Comment by Tlaltecuhtli** [comment]
you might have thought you made a fine and balanced chem but you forgot about slimepeople which heal on toxin dmg, checkmate

**Comment by Rowelll** [comment]
> * Salbu has been added to sleepers as a T3 upgrade chemical (hehe)

~~who upgrades sleepers in shuttles~~

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm (hunk 1 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: Sorta ;No (even when OD'd)

It will have the potential to go up 1 per tick. However, it's adjusted for oxygen healed.

Let's say you're at 100 `current_cycle`. That means you can heal 250 Oxydamage. If you only have 50 oxydamage though, you'll only take 20 toxin damage (then cycle 101 you'll prob just take 0.5 since you've healed all oxydamage).

If you're OD'd though you will take 100 toxdamage if you somehow managed to have enough to get to that point.

**Comment by ExcessiveUseOfCobblestone** [comment]
> you might have thought you made a fine and balanced chem but you forgot about slimepeople which heal on toxin dmg, checkmate

it's just tox damage right? Seems like a neat side effect tbh. just beat their skull in lol

**Comment by ATH1909** [comment]
https://www.youtube.com/watch?v=JaotiOp45dg

Oh boy, I can't wait to die from a choking+ARDS virus with much less available counterplay once chem is bombed.

Seriously, though, just... why? Why would I want to turn my oxygen damage into the much more annoying to heal (especially since charcoal will purge perf) toxin damage? just breathe lmao

Just remove oxygen deprivation kits if you want to make them filled with godawful chems like your revised version of perf.

**Comment by ExcessiveUseOfCobblestone** [comment]
I think I addressed this in the previous PR but you **shouldn't** use this chem for basic oxygen healing, that's the point and why it's in the EMERGENCY oxygen deprivation kits. It should be used for emergencies only and dripfed to ensure you get the max benefit.

It also pairs well with epi since it heals both oxy and tox while in crit.

**Comment by ATH1909** [comment]
> I think I addressed this in the previous PR but you **shouldn't** use this chem for basic oxygen healing **because** CPR exists, that's the point and why it's in the EMERGENCY oxygen deprivation kits. It should be used for emergencies only and dripfed to ensure you get the max benefit.
> 
> It also pairs well with epi since it heals both oxy and tox while in crit.

Shouldn't _CPR_ be the emergency oxygen healing option?

**Comment by ExcessiveUseOfCobblestone** [comment]
It still is if you don't have any meds. It's not worth doing if you're still being hit with O2 damage nor does it fix losebreath.

CPR's pro is that it doesn't need chems and can be done by anyone but it's con is that it sucks and the person doing it is "paralyzed" during the process.

Perf's pro is that it will easily fix you if you're taking a bunch of oxyloss but it's con is that it converts 40% healed to toxin damage plus 0.5 tox per tick when it's not. The latter can be mitigated by appropriate dosing/ drip feeding. More importantly, you can multitask with it unlike CPR.

In terms of combining with other meds, Epi with CPR will only heal 0.5 dptick (oxy) whereas Epi with Perf will heal 1 dptick (oxy/tox), and that's not even considering perf fully removes 60% of oxydamage it heals with no downside.

In short, yes it should be and it will still be the final final option in an emergency if you don't have perf available, but It may not be sufficient as one might would expect from a last resort option.

**Comment by PKPenguin321** [comment]
This sounds awful since toxin damage is easily the most annoying type
Can definitely see the meta just being to never, ever use this chem

**Comment by nightred** [comment]
this change is interesting in that it is not a perfect drug, the idea that it is helpful and harmful is interesting system to play with. I believe this could have a positive effect on game play and would help medical be more a active part of the station.

**Comment by ExcessiveUseOfCobblestone** [comment]
> This sounds awful since toxin damage is easily the most annoying type
> Can definitely see the meta just being to never, ever use this chem

I would argue cloning/brain are more annoying but everything is rocket science when the metric is based on brute/burn number-sliders.

**Comment by YakumoChen** [comment]
Cloning damage has some pretty extremely rare sources (besides cloning itself), brain damage is similarly rare and doesn't actually contribute towards your HP.

**Comment by PKPenguin321** [comment]
Yeah plus it's pretty clear that I was talking about the four main damage types and going "UHM TECHNICALLY EYE DAMAGE IS A TYPE" is not really addressing anything I said

**Comment by ExcessiveUseOfCobblestone** [comment]
I mean they're on the top of a mob's VV, I wasn't trying to gotcha which I figured would be obvious since I went on to mention brute/burn after.

Anyways I don't want to draw the conclusion that because brute/burn are piss easy to heal that means I should change toxin to something else. The chem is for situational healing where CPR won't be as effective as I said earlier and it can be stacked with other chems to negate any downside of toxin, regardless of how "annoying" you believe toxin is to heal.

If CPR is sufficient for literally every situation then we should look at buffing oxyloss damage now that the medicines aren't braindead healing and/or require serious resources.

**Comment by imsxz** [comment]
just use antitoxin to heal it you fuckin troglodytes it only removes toxin reagents from your blood and it has a piss easy chem recipe holy moly lads can you go 1 fuckin change without focusing on the negatives

@athath this is better for fighting ARDS than salb Cobby said it and I’m too lazy to fact check

**Comment by ExcessiveUseOfCobblestone** [comment]
what is ards
